src/common/QueryHandler.ts

```diff
@@ -191,11 +210,46 @@ const pierceHandler = createInternalQueryHandler({
   },
 });
 
-const builtInHandlers = new Map([
-  ['aria', ariaHandler],
-  ['pierce', pierceHandler],
+const xpathHandler = internalizeCustomQueryHandler({
+  queryOne: (element, selector) => {
+    const doc = element.ownerDocument || document;
+    const iterator = doc.evaluate(
+      selector,
+      element,
+      null,
+      XPathResult.ORDERED_NODE_ITERATOR_TYPE
```

Comment:
should we use FIRST_ORDERED_NODE_TYPE here? it seems like ORDERED_NODE_ITERATOR_TYPE would be fetching all matching elements? (not sure exactly what that means in the xpath impl)